.github/workflows/inspektor-gadget.yml

```diff
@@ -1,45 +1,92 @@
-name: Compile Inspektor Gadget
+name: Inspektor Gadget CI
 on:
   push:
 
+# Jobs are given a level in a comment.
+# Jobs of the same level run in parallel.
+# Jobs of level N depend of, at least, one job on level N - 1.
 jobs:
-
-  build:
-    name: Build
+  checkout-code:
+    name: Checkout code
+    # level: 0
     runs-on: ubuntu-latest
-    env:
-      CONTAINER_REPO: ${{ secrets.CONTAINER_REPO }}
     steps:
+    - name: Check out code
+      uses: actions/checkout@v1
+    - name: Create inspektor-gadget source code archive.
+      # *-artifact does not maintain permission:
+      # https://github.com/actions/upload-artifact#permission-loss
+      # So, we will tar the about to be upload artifact and untar once
+      # downloaded.
+      run: |
+        cd /home/runner/work/inspektor-gadget
+        tar -cf inspektor-gadget.tar inspektor-gadget
+    - name: Add inspektor-gadget source as artifact.
+      uses: actions/upload-artifact@master
+      with:
+        name: inspektor-gadget-tar
+        path: /home/runner/work/inspektor-gadget/inspektor-gadget.tar
 
-    - name: Set up Go 1.17
+  documentation-checks:
+    name: Documentation checks
+    # level: 1
+    needs: checkout-code
+    runs-on: ubuntu-latest
+    steps:
+    - name: Setup go 1.17
       uses: actions/setup-go@v1
       with:
         go-version: 1.17
       id: go
+    - name: Cache go 1.17
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.cache/go-build
+          ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+    - name: Get inspektor-gadget from artifact.
+      uses: actions/download-artifact@v2
+      with:
+        name: inspektor-gadget-tar
+        path: /home/runner/work/inspektor-gadget/
+    - name: Check if generated files are updated
+      run: |
+        tar -xf /home/runner/work/inspektor-gadget/inspektor-gadget.tar
+        cd inspektor-gadget
 
-    - name: Check out code
-      uses: actions/checkout@v1
+        make manifests generate generate-documentation
+        git diff --exit-code HEAD --
 
-    - name: Cache external deps
+  lint:
+    name: Lint (gofumpt)
```

Comment:
I'd expect more linters to be enabled over time, so I wouldn't necessarily mention `gofumpt` here.